client/src/actions/grantsMetadata.ts

```diff
@@ -58,51 +59,66 @@ export const fetchGrantData =
       signer
     );
 
-    let grant: {
+    let project: {
       metadata: {
         protocol: number;
         pointer: string;
       };
     };
+
     try {
-      grant = await projectRegistry.projects(id);
+      project = await projectRegistry.projects(id);
     } catch (e) {
       // FIXME: dispatch contract interaction error
-      console.log(e);
+      console.error(e);
       return;
     }
 
-    if (grant === null) {
+    // FIXME: the contract will always return a project,
+    // check one field to know if it exists.
+    if (project === null) {
       // FIXME: dispatch "not found"
-      console.error("grant not found");
+      console.error("project not found");
       return;
     }
 
-    const cachedMetadata = state.grantsMetadata[id]?.metadata;
-    if (
-      cachedMetadata !== undefined &&
-      cachedMetadata.uri === grant.metadata.pointer
-    ) {
-      dispatch(grantMetadataFetched(cachedMetadata));
-      return;
+    const chunks = [];
```

Comment:
This function is getting pretty huge, maybe we could consolidate it in to a few testable functions? Wdyt?